substrate-node/pallets/pallet-smart-contract/src/tests.rs

```diff
@@ -2369,6 +2375,584 @@ fn test_create_node_contract_with_solution_provider_fails_if_not_approved() {
     });
 }
 
+// SERVICE CONTRACT TESTS //
+// ---------------------- //
+
+#[test]
+fn test_service_contract_create_works() {
+    new_test_ext().execute_with(|| {
+        run_to_block(1, None);
+        create_service_consumer_contract();
+
+        assert_eq!(
+            get_service_contract(),
+            SmartContractModule::service_contracts(1).unwrap(),
+        );
+
+        let our_events = System::events();
+        assert_eq!(!our_events.is_empty(), true);
```

Comment:
Nitpick: not really necessary as the following assert will fail too if the events are empty.